airbyte-ci/connectors/metadata_service/lib/metadata_service/validators/metadata_validator.py

```diff
@@ -206,27 +206,31 @@ def validate_docker_image_tag_is_not_decremented(
     return True, None
 
 
-def validate_release_candidate_has_rc_suffix_in_version(
+def validate_rc_suffix_and_rollout_configuration(
     metadata_definition: ConnectorMetadataDefinitionV0, _validator_opts: ValidatorOptions
 ) -> ValidationResult:
     # Bypass validation for pre-releases
     if _validator_opts and _validator_opts.prerelease_tag:
         return True, None
 
-    is_release_candidate = get(metadata_definition, "data.releases.isReleaseCandidate")
-    if not is_release_candidate:
-        return True, None
-
     docker_image_tag = get(metadata_definition, "data.dockerImageTag")
     if docker_image_tag is None:
         return False, "The dockerImageTag field is not set."
     try:
         parsed_version = semver.VersionInfo.parse(docker_image_tag)
+        is_prerelease = parsed_version.prerelease is not None
```

Comment:
It looks like semver treats the `rc` suffix as a prerelease. Meaning that from the perspective of some of the code (like here) release candidates are a subset of prereleases.

This does make sense, but since our handling of what we consider prereleases is so different from release candidates, I wonder if we want to make a more obvious distinction between them in the code. I think maybe that would mean having dedicated functions for `is_prerelease` and `is_release_candidate` instead of using the semver output directly to define a prerelease.

Reply:
I slightly reworked the code to make the logic more readable. Thanks for the feedback.